src/context/auth-context.js

```diff
@@ -0,0 +1,54 @@
+import React, { useContext, createContext, useState } from "react";
+import axios from "axios";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [user, setUser] = useState({});
+  const [status, setStatus] = useState("");
+  const [newUser, setNewUser] = useState({});
```

Comment:
Small suggestion , I know this will make you refactor your code.
But we could have used ```useReducer``` here.